app/controllers/password_controller.rb

```diff
@@ -33,6 +35,19 @@ def email_sent
   private
 
   def password_params
-    params.require(:user).permit(:password, :password_confirmation)
+    params.require(:user).permit(:password)
+  end
+
+  def aws_email_service
+    edit_users_password_link = edit_users_password_path(email: @user.email, reset_token: @user.reset_token)
+
+    EmailService.new(client: aws_ses_client, 
+                      users: @user, 
+                      type: "password_reset_email",
+                      link: edit_users_password_link)
+  end
+
+  def aws_ses_client
+    Aws::SES::Client.new
   end
```

Comment:
You probably want to move this configuration into your `EmailService` as defaults. Then instead of `aws_email_service.send_email!` you're just calling `send_reset_password_email` or whatever.

```
  def send_reset_password_email
    edit_users_password_link = edit_users_password_path(email: @user.email, reset_token: @user.reset_token)

    EmailService.new(users: @user, type: "password_reset_email", link: edit_users_password_link).send_email!
  end
```

Reply:
I was thinking about abstracting EmailService and creating a subclass called AwsEmailService. Then those defaults would be in that but I think its too early to go down that road

Reply:
Yeah, probably too early to subclass the email service. And even then, implementation details are probably less important than business reasons, i.e. you'd probably be looking at 
```
class CampaignEmailService < EmailService
...
def initialize(client: SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY']), ...)
```
rather than specific `AwsEmailService` or `HubspotEmailService`. But I could be wrong, depending on your app.

But regardless, I still think that the `User` class shouldn't really be concerned with the `EmailService`'s configuration setup as using the AWS client is probably not User specific.